arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -733,51 +735,84 @@ public void tracePostExecution(MessageFrame frame, Operation.OperationResult ope
         this.state().processingPhase == TX_EXEC,
         "There can't be any execution if the HUB is not in execution phase");
 
-    final long gasCost = operationResult.getGasCost();
     final TraceSection currentSection = state.currentTxTrace().currentSection();
 
-    final short exceptions = this.pch().exceptions();
+    compareLineaAndBesuGasCosts(frame, operationResult);
 
-    final boolean memoryExpansionException = Exceptions.memoryExpansionException(exceptions);
-    final boolean outOfGasException = Exceptions.outOfGasException(exceptions);
-    final boolean unexceptional = Exceptions.none(exceptions);
-    final boolean exceptional = Exceptions.any(exceptions);
-
-    // NOTE: whenever there is an exception, a context row
-    // is added at the end of the section; its purpose is
-    // to update the caller / creator context with empty
-    // return data.
-    ///////////////////////////////////////////////////////
-    if (exceptional) {
+    /*
+     * NOTE: whenever there is an exception, a context row
+     * is added at the end of the section; its purpose is
+     * to update the caller / creator context with empty
+     * return data.
+     */
+    if (isExceptional()) {
       this.currentTraceSection()
           .addFragments(ContextFragment.executionProvidesEmptyReturnData(this));
       this.squashCurrentFrameOutputData();
       this.squashParentFrameReturnData();
     }
 
-    // Setting gas cost IN MOST CASES
-    // TODO:
-    //  * complete this for CREATE's and CALL's
-    //    + are we getting the correct cost (i.e. excluding the 63/64-th's) ?
-    //  * make sure this aligns with exception handling of the zkevm
-    //  * write a method `final boolean requiresGasCost()` (huge switch case)
-    if ((!memoryExpansionException & outOfGasException) || unexceptional) {
-      currentSection.commonValues.gasCost(gasCost);
-      currentSection.commonValues.gasNext(
-          unexceptional ? currentSection.commonValues.gasActual - gasCost : 0);
-    } else {
-      currentSection.commonValues.gasCost(
-          0); // TODO: fill with correct values --- make sure this works in all cases
-      currentSection.commonValues.gasNext(0);
-    }
-
```

Comment:
Main point: computing the gas cost and gas next _after_ instruction processing by Besu leads to wrong results: the stack items were already popped, memory was expanded, addresses may or may not have been warmed at that stage ... the pricing will be off.

We do these computations (that belong to the common values) in the `CommonValuesFragment` now. It is therefore triggered with every `TraceSection`.